### PR TITLE
added two data types javascript questions

### PR DESCRIPTION
**Pull request by johansantana**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #XXXXX

### CODE REVIEW

File: src/data/javascript-quiz.ts [hunk 1]
```diff
@@ -1189,7 +1189,27 @@ const javascriptQuiz = [
     Distractor2: "Integer",
     Distractor3: "Boolean",
     Explanation:
-      "In JavaScript, typeof null is an object.This result of typeof null is actually a bug in the language.",
+      "In JavaScript, typeof null is an object. This result of typeof null is actually a bug in the language.",
+    Link: "https://www.freecodecamp.org/news/javascript-typeof-how-to-check-the-type-of-a-variable-or-object-in-js/",
+  },
+  {
+    Question: "In JavaScript, what will 'typeof []' return:",
```
Comment (jdwilkin4) [inline src/data/javascript-quiz.ts:1196]:
```suggestion
    Question: "In JavaScript, what will 'typeof []' return?",
```

File: src/data/javascript-quiz.ts [hunk 2]
```diff
@@ -1189,7 +1189,27 @@ const javascriptQuiz = [
...
+    Answer: "Object",
+    Distractor1: "Null",
+    Distractor2: "Array",
+    Distractor3: "Undefineed",
```
Comment (jdwilkin4) [inline src/data/javascript-quiz.ts:1200]:
```suggestion
    Distractor3: "Undefined",
```

File: src/data/javascript-quiz.ts [hunk 3]
```diff
@@ -1189,7 +1189,27 @@ const javascriptQuiz = [
...
+    Explanation:
+      "In JavaScript, there is no array type. Everything that is not a primitive value is an object.",
```
Comment (jdwilkin4) [inline src/data/javascript-quiz.ts:1202]:
```suggestion
      "In JavaScript, everything that is not a primitive value is an object.",
```

File: src/data/javascript-quiz.ts [hunk 4]
```diff
@@ -1189,7 +1189,27 @@ const javascriptQuiz = [
...
+    Link: "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures",
+  },
+  {
+    Question: "In JavaScript, what will 'typeof NaN' return:",
+    Answer: "Number",
+    Distractor1: "ReferenceError",
+    Distractor2: "Null",
+    Distractor3: "NaN",
+    Explanation:
+      "In JavaScript, the NaN (not-a-number) is still a numeric data type, but it is undefined as a real number.",
```
Comment (jdwilkin4) [inline src/data/javascript-quiz.ts:1212]:
```suggestion
      "In JavaScript, NaN (not-a-number) is still a numeric data type, but it is undefined as a real number.",
```

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Please review the requested changes.
Once those changes have been implemented, then we can merge your PR into main. 👍

**Comment by johansantana** [comment]
> Thank you for your contribution!
> 
> Please review the requested changes. Once those changes have been implemented, then we can merge your PR into main. 👍

Done!

**Review by jdwilkin4 — APPROVED** [review]
Congrats on your first contribution to our repository. 🎉
We look forward to future contributions. 👍

**Pull request merged**